Update for networkx 2.3 --> 2.4, minor safety fixes in obo.py

srcoulombe opened the pull request:
== Update for NetworkX 2.3 --> 2.4 ==

As per the release notes for NetworkX2.4 (https://networkx.github.io/documentation/stable/release/release_2.4.html#deprecations)

G.node –> use G.nodes

The nx.Graph class's .node attribute is depreciated in favour of the .nodes attribute (which was in previous versions of NetworkX). No changes in functionality though. 

Changes: All <DiGraph>.node calls were replaced with <DiGraph>.nodes.

== minor file IO safety fix ==

The ontology function in goenrich/obo.py had non-descript behaviour for inputs being non-file path strings. It also delayed the _io.TextIOWrapper.close() call, which seemed unintentionally/unnecessarily risky.

Changes: f = open(file) and calling f.close() in goenrich/obo.py were replaced with the safer context-manager way (with open(file, mode) as f:...). Also replaced 

else:
        f = file
        we_opened_file = False

srcoulombe committed:
Update for networkx 2.3 --> 2.4, minor safety fixes in obo.py

== Update for NetworkX 2.3 --> 2.4 ==

As per the release notes for NetworkX2.4 (https://networkx.github.io/documentation/stable/release/release_2.4.html#deprecations)

G.node –> use G.nodes

The nx.Graph class's .node attribute is depreciated in favour of the .nodes attribute (which was in previous versions of NetworkX). No changes in functionality though. 

Changes: All <DiGraph>.node calls were replaced with <DiGraph>.nodes.

== minor file IO safety fix ==

The ontology function in goenrich/obo.py had non-descript behaviour for inputs being non-file path strings. It also delayed the _io.TextIOWrapper.close() call, which seemed unintentionally/unnecessarily risky.

Changes: f = open(file) and calling f.close() in goenrich/obo.py were replaced with the safer context-manager way (with open(file, mode) as f:...). Also replaced 

else:
        f = file
        we_opened_file = False

srcoulombe commented:
This would fix https://github.com/jdrudolph/goenrich/issues/14

jdrudolph commented:
Hi!
Amazing! Thanks a lot for you patch! I'm definitely interested in merging this. Maybe we could discuss/address the following points before:

- [ ] I would like to remain somewhat backwards compatible. If possible `networkx` version should be checked and the appropriate api should be used.
- [ ] Unit tests in CI should pass. Let me know if something is unclear here.
- [ ] The diff makes it look like the entire file changed. Could there be some issue with line endings that trips git up?
- [ ] Could you describe how you could use either a file path or a file object as argument for `obo`? I was aiming for an api as convenient as `pandas` where you can do either `pd.read_csv('file.csv')` or you can do `with open('file.csv') as f: pd.read_csv(f)`. That is why I came up with this code.

Let me know what you think of this.
Best, Jan

srcoulombe committed:
removed dependency on readme()

Caused issues when trying to install from a different directory.

srcoulombe committed:
Changes to be committed:
	modified:   goenrich/enrich.py
	modified:   goenrich/export.py
	modified:   goenrich/obo.py

Changes:
	1. 	added compatibility with networkx 2.4+ (after the nx.Graph class and subclasses have had their `.node` attribute replaced with `.nodes`)
		checks the version of the networkx module that was imported.
			if < 2.4, then the `.node` attribute is added as an alias to the `.nodes` attribute for the nx.Graph class and its subclasses
			else, no change is made since the `.node` attribute already exists

srcoulombe committed:
Merge branch 'rework'

srcoulombe committed:
On branch master
Your branch is ahead of 'origin/master' by 2 commits.
 Changes to be committed:
	modified:   goenrich/enrich.py
	modified:   goenrich/export.py
	modified:   goenrich/obo.py
	modified:   goenrich/tests/test_enrichment.py
	modified:   goenrich/tests/test_propagation.py

Summary:
	Make goenrich compatible with networkx >= 2.4 as well as < 2.4 legacy versions.
	goenrich currently makes calls to networkx.DiGraphs' .node property, which has been depreciated in networkx 2.4.0 in favor of the .nodes property.
	This was done by checking the imported networkx module's version, and dynamically aliasing the .nodes property with a .node property if need be.

jdrudolph commented:
Addressed in PR #20 and release 1.12
